docs/Using Fleet/GitOps.md

```diff
@@ -322,7 +322,7 @@ software:
 
 #### app_store_apps
 
-- `app_store_id` is the ID of the Apple App Store app. You can find this at the end of the app's App Store URL. For example, "Bear - Markdown Notes" URL is "https://apps.apple.com/us/app/bear-markdown-notes/id1016366447" and the `app_store_id` is `1016366447` (default: `0`).
+- `app_store_id` is the ID of the Apple App Store app. You can find this at the end of the app's App Store URL. For example, "Bear - Markdown Notes" URL is "https://apps.apple.com/us/app/bear-markdown-notes/id1016366447" and the `app_store_id` is `1016366447`.
```

Comment:
@noahtalerman Not sure why do we have "(default: `0`)" here. Is that mistake?

Reply:
Hey @marko-lisica, I think Rachael Shaw and I made an assumption here.

Now that it's a string, would the default be an empty string?

Reply:
Not sure if I get why we need a default value here? If someone provides YAML like this?

```
software:
  app_store_apps:
   - app_store_id: '1091189122'
   - app_store_id: 
```

Reply:
That's right. Hey @dantecatalfamo what happens if the user does that? ^

Reply:
@noahtalerman Good question, I'll test that and get back to you.

What should it do from a product POV?

Reply:
@marko-lisica What should happen if the user does that?

Reply:
Hey @dantecatalfamo! Thanks for testing.

Also, we don't validate that the ID is a valid App Store app ID right?

I think ignoring an empty and invalid App Store app ID is ok. As a user who added an empty or invalid ID, I'll see that the App Store app wasn't added. Then I'll double check the ID.

If we learn that users are tripping over these I think we come back and add validation.

Reply:
Hey @dantecatalfamo, just checking, did you get a chance to test this? https://github.com/fleetdm/fleet/pull/20875#discussion_r1702121233

Reply:
Hey @noahtalerman, it looks like currently the backend will throw an error. Do we want to create a client-side error telling the user to remove the empty key or just ignore the entry?

Reply:
Hey @noahtalerman, if the ID is empty, it will return an error to the user saying `software app store id required`

Reply:
The code to check and return the error already existed, but I'm adding a new test right now to make sure this will always be the case 🙂

Reply:
Nice!